x-pack/plugins/security_solution/server/endpoint/services/actions/create/index.ts

```diff
@@ -51,31 +54,45 @@ const commandToFeatureKeyMap = new Map<ResponseActionsApiCommandNames, FeatureKe
 ]);
 
 const returnActionIdCommands: ResponseActionsApiCommandNames[] = ['isolate', 'unisolate'];
+type CreateActionPayload = TypeOf<typeof ResponseActionBodySchema> & {
+  command: ResponseActionsApiCommandNames;
+  user?: ReturnType<AuthenticationServiceStart['getCurrentUser']>;
+  rule_id?: string;
+  rule_name?: string;
+  error?: string;
+};
+
+interface CreateActionMetadata {
+  casesClient?: CasesClient;
+  minimumLicenseRequired?: LicenseType;
+  enableActionsWithErrors?: boolean;
+}
 
-export class ActionCreateService {
-  constructor(private esClient: ElasticsearchClient, private endpointContext: EndpointAppContext) {}
+export const actionCreateService = (
+  esClient: ElasticsearchClient,
+  endpointContext: EndpointAppContext,
+  licenseService: LicenseService
```

Comment:
Why do you need pass in the `licenseService` here? Its already available in `endpointContext`

Reply:
Good catch! My previous implementation was checking the license on service setup, and I think it was needed then. But now we check it during the action call so we already have it in context. Anyway, I'll change it to be taken from `endpointContext`. Thanks Paul!